File: algosort/bubble/bubblesort.py
unsortedArray = [2018, 1998, 1986, 2020, 2006]


def bubbleSort(arrayToSort):
    arrayIsSorted = False

    while arrayIsSorted is False:
        arrayIsSorted = True

        for i in range(len(arrayToSort) - 1):
            if (arrayToSort[i] > arrayToSort[i + 1]):
                tempLeftValue = arrayToSort[i]
                arrayToSort[i] = arrayToSort[i + 1]
                arrayToSort[i + 1] = tempLeftValue
                arrayIsSorted = False

    return arrayToSort


print(bubbleSort(unsortedArray))

File: algosort/bubble/bubblesort.js
const unsortedArray = [2020, 1998, 2018, 1986, 2006]

const bubbleSort = array => {
  const arrayLength = array.length
  let isSwapped

  do {
    isSwapped = false

    for (let i = 0; i < arrayLength; i++) {
      if (array[i] > array[i + 1]) {
        const tempLeftValue = array[i]
        array[i] = array[i + 1]
        array[i + 1] = tempLeftValue

        isSwapped = true
      }
    }
  } while (isSwapped)

  return array
}

console.log(bubbleSort(unsortedArray))
// [ 1986, 1998, 2006, 2018, 2020 ]
